cmake/onnx.cmake

```diff
@@ -0,0 +1,34 @@
+PROTOBUF_GENERATE_CPP(PROTO_SRCS PROTO_HDRS
+    ${ONNXIR_ROOT}/onnx/onnx.proto
+    ${ONNXIR_ROOT}/onnx/onnx-operators.proto
+)
+
+file(GLOB_RECURSE onnx_src
+    "${ONNXIR_ROOT}/onnx/*.h"
+    "${ONNXIR_ROOT}/onnx/*.cc"
+)
+
+file(GLOB_RECURSE onnx_exclude_src
+    "${ONNXIR_ROOT}/onnx/onnx.pb.h"
+    "${ONNXIR_ROOT}/onnx/onnx.pb.cc"
+    "${ONNXIR_ROOT}/onnx/onnx-operators.pb.h"
+    "${ONNXIR_ROOT}/onnx/onnx-operators.pb.cc"
+)
+list(LENGTH onnx_exclude_src list_len)
+if (list_len GREATER 1)
+    list(REMOVE_ITEM onnx_src ${onnx_exclude_src})
+endif()
+
+add_library(onnxir ${PROTO_SRCS} ${PROTO_HDRS} ${onnx_src})
+
+if (WIN32)
+    target_compile_options(onnxir PRIVATE
```

Comment:
A friendly reminder that you might need more than these warnings, thanks to protobuf which is a huge pot of warning-generator. More specifically, this list might help:

https://github.com/caffe2/caffe2/blob/3eb636c0edc9337fb348ec554e366b1020604c68/cmake/MiscCheck.cmake#L110-L126

Reply:
currently only the warnings that specified here need to be taken care of, as I can get a successful build now. We may add more when needed?